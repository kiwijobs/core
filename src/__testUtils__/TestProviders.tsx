import React, { FC, Suspense } from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../theme';

export const TestProviders: FC<{}> = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <Suspense fallback="mocked loading...">{children}</Suspense>
    </ThemeProvider>
  );
};

export const wrapper = TestProviders; // React Testing Library friendly name.
