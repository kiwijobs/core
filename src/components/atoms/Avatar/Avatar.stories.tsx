import React from 'react';
import { storiesOf } from '@storybook/react';
import { Box, Flex } from '../../quarks';
import { Avatar } from './Avatar';

storiesOf('Atoms/Avatar', module).add('Various', () => (
  <>
    <Box display="inline-block" p={3}>
      <Avatar src="https://placebear.com/200/400" />
    </Box>
    <Box display="inline-block" p={3}>
      <Avatar size="60px" src="https://placebear.com/200/401" />
    </Box>
    <Box display="inline-block" p={3}>
      <Avatar size="72px" src="https://placebear.com/200/402" />
    </Box>
    <Box display="inline-block" p={3}>
      <Avatar size="80px" src="https://placebear.com/200/403" />
    </Box>
    <Flex alignItems="center" p={3}>
      <Avatar size="40px" src="https://placebear.com/200/404" />
      <Box fontWeight="bold" ml={2}>
        Pan Niedźwiedź
      </Box>
    </Flex>
  </>
));
