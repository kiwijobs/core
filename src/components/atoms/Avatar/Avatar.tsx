import React, { forwardRef } from 'react';
import { Image, ImageProps } from '../../quarks';

export const Avatar = forwardRef<HTMLImageElement, ImageProps>(({ sx, ...props }, ref) => (
  <Image
    ref={ref}
    sx={{
      size: '48px',
      flex: '0 0 auto',
      bg: '4',
      borderRadius: '50%',
      objectFit: 'cover',
      ...sx,
    }}
    {...props}
  />
));
