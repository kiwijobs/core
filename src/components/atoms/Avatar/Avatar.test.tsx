import React from 'react';
import { wrapper } from '../../../__testUtils__';
import { Avatar } from './Avatar';
import { render } from '@testing-library/react';

describe('Avatar', () => {
  it('Matches snapshot', () => {
    expect(render(<Avatar src="/img.jpg" />, { wrapper }).container).toMatchSnapshot();
    expect(render(<Avatar src="/img.jpg" size="32px" />, { wrapper }).container).toMatchSnapshot();
    expect(render(<Avatar src="/img.jpg" size={60} />, { wrapper }).container).toMatchSnapshot();
  });
});
