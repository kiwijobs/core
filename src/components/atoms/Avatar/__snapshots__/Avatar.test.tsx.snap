// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Matches snapshot 1`] = `
.c0 {
  width: 48px;
  height: 48px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  background-color: rgb(228,230,237);
  border-radius: 50%;
  object-fit: cover;
}

<div>
  <img
    class="c0 "
    src="/img.jpg"
  />
</div>
`;

exports[`Avatar Matches snapshot 2`] = `
.c0 {
  width: 48px;
  height: 48px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  background-color: rgb(228,230,237);
  border-radius: 50%;
  object-fit: cover;
  width: 32px;
  height: 32px;
}

<div>
  <img
    class="c0 "
    src="/img.jpg"
  />
</div>
`;

exports[`Avatar Matches snapshot 3`] = `
.c0 {
  width: 48px;
  height: 48px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  background-color: rgb(228,230,237);
  border-radius: 50%;
  object-fit: cover;
  width: 60px;
  height: 60px;
}

<div>
  <img
    class="c0 "
    size="60"
    src="/img.jpg"
  />
</div>
`;
